src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2727,7 +2727,7 @@ object Types {
 
         case tp @ AnnotatedType(annot, underlying) =>
           val underlying1 = this(underlying)
-          if (underlying1 eq underlying) tp else underlying1
+          if (underlying1 eq underlying) tp else tp.derivedAnnotatedType(annot, underlying1)
```

Comment:
Nice :-)

Out of curiosity, I had a look at the implementation of `mapOver(annot)`, which is implemented using `TreeTypeMap`, which extends `tpd.TreeMap`. Now it seems to me that this does the same as `TreeTransform`, so I wonder why there are two different concepts here? [This has (almost) nothing to do with this PR, sorry for hijacking this thread with my question ;-)]